base/float.jl

```diff
@@ -137,6 +137,68 @@ i.e. the maximum integer value representable by [`exponent_bits(T)`](@ref) bits.
 """
 function exponent_raw_max end
 
+"""
+IEEE 754 definition of the minimum exponent.
+"""
+ieee754_exponent_min(::Type{T}) where {T<:IEEEFloat} = Int(1 - exponent_max(T))::Int
+
+exponent_min(::Type{Float16}) = ieee754_exponent_min(Float16)
```

Comment:
Do these functions need to exist?

Reply:
You mean why is `exponent_min` necessary in addition to `ieee754_exponent_min`?

The idea was to have `exponent_min` be consistent with functions like `exponent_max`, in that it only has methods that are strictly required. To be honest, I don't really understand the significance of that, I was just trying to follow the discussion in #50654, the "incorrectly switches the position / ordering of `Union` and `Type`" part, in particular.

Apart from that the formula in the definition of `ieee754_exponent_min` is defined in the IEEE 754 standard, and it's only known to be valid for IEEE 754 floating-point. So theoretically we could have some non-IEEE floating-point types with different definitions of `exponent_min`.